packages/autocomplete-core/src/__tests__/getInputProps.test.ts

```diff
@@ -1165,9 +1165,151 @@ describe('getInputProps', () => {
+      });
+
+      inputElement.focus();
+
+      expect(onStateChange).toHaveBeenLastCalledWith({
+        prevState: expect.anything(),
+        state: expect.objectContaining({
+          activeItemId: 1,
+          isOpen: true,
+        }),
+      });
+    });
+  });
+
+  describe('onBlur', () => {
+    test('resets activeItemId and isOpen', () => {
+      const onStateChange = jest.fn();
+      const { inputElement } = createPlayground(createAutocomplete, {
+        onStateChange,
+        defaultActiveItemId: 1,
+        openOnFocus: true,
+      });
+      inputElement.focus();
+
+      inputElement.blur();
+
+      expect(onStateChange).toHaveBeenLastCalledWith({
+        prevState: expect.anything(),
+        state: expect.objectContaining({
+          activeItemId: null,
+          isOpen: false,
+        }),
+      });
+    });
+    test('does not reset activeItemId and isOpen on touch devices', () => {
+      const environment = {
+        ...global,
+        ontouchstart: () => {},
+      };
+      const onStateChange = jest.fn();
+      const { inputElement } = createPlayground(createAutocomplete, {
+        environment,
+        onStateChange,
+        defaultActiveItemId: 1,
+        openOnFocus: true,
+      });
+      inputElement.focus();
+
+      inputElement.blur();
+
+      expect(onStateChange).toHaveBeenLastCalledWith({
+        prevState: expect.anything(),
+        state: expect.objectContaining({
+          activeItemId: 1,
+          isOpen: true,
+        }),
+      });
+    });
+  });
 
-  describe('onBlur', () => {});
+  describe('onClick', () => {
+    test('is a noop when the input is not focused', () => {
+      const onStateChange = jest.fn();
+      const { inputElement } = createPlayground(createAutocomplete, {
+        onStateChange,
+      });
 
-  describe('onClick', () => {});
+      inputElement.click();
+
+      expect(onStateChange).toHaveBeenCalledTimes(0);
+    });
+
+    test('is a noop when panel is already open', () => {
+      const onStateChange = jest.fn();
+      const { inputElement } = createPlayground(createAutocomplete, {
+        onStateChange,
+        openOnFocus: true,
+        initialState: {
+          isOpen: true,
+        },
+      });
+
+      inputElement.focus();
+
+      // Clear mock information set by the `focus` event
+      onStateChange.mockClear();
+
+      inputElement.click();
+
+      expect(onStateChange).toHaveBeenCalledTimes(0);
+    });
+
+    test('sets activeItemId and isOpen when the input is focused and the panel closed', () => {
+      const onStateChange = jest.fn();
+      const { inputElement } = createPlayground(createAutocomplete, {
+        onStateChange,
+        defaultActiveItemId: 1,
+      });
+
+      inputElement.focus();
+
+      inputElement.click();
+
+      expect(onStateChange).toHaveBeenLastCalledWith({
+        prevState: expect.anything(),
+        state: expect.objectContaining({
+          activeItemId: 1,
+          isOpen: false,
```

Comment:
I don't think `onClick` can change `isOpen`, either the panel is already open or it should not be opened. Thought ?

Reply:
You either need to have an initial query or to provide `openOnFocus` for this test to work with `isOpen === true`.

Reply:
In either case, if there is an initial query or if `openOnFocus` is true then the panel is already opened so the `onClick` handler is a noop.

Reply:
You can trigger a keyboard event with the `Escape` key to close the panel and keep the input focused.

Reply:
It's the info that I was missing. Thanks !